geode-core/src/main/java/org/apache/geode/internal/net/SocketCreator.java

```diff
@@ -362,38 +362,36 @@ public SSLEngine createSSLEngine(String hostName, int port, boolean clientSocket
   }
 
   @VisibleForTesting
-  void configureSSLEngine(SSLEngine engine, String hostName, int port, boolean clientSocket) {
-    SSLParameters parameters = engine.getSSLParameters();
-    boolean updateEngineWithParameters = false;
+  void configureSSLEngine(final SSLEngine engine, final String hostName, final int port,
+      final boolean clientSocket) {
+    engine.setUseClientMode(clientSocket);
+    final SSLParameters parameters = engine.getSSLParameters();
+    configureSSLParameters(parameters, hostName, port, clientSocket);
+    engine.setSSLParameters(parameters);
```

Comment:
Do you see any reason for the previous code trying to avoid invoking `engine.setSSLParameters` when nothing has changed?  (the `updateEngineWithParameters` flag)

Reply:
I couldn't find any good reason. I think It was an attempt to avoid the issues reported in [GEODE-8419](https://issues.apache.org/jira/browse/GEODE-8419). But looking at the sources setting all these parameters via the `SSLParameters` is the safest approach. Alternatively we could get and set the SSLParameters around each parameter.

Reply:
I've reviewed the [PR for GEODE-8419](https://github.com/apache/geode/pull/5465) and it looks like the idea in both this method and in `configureClientSSLSocket()` was to set parameters at most once, and to avoid setting them at all if they didn't need to be modified.

I wondered if setting them more aggressively e.g. unconditionally, had some untoward effect but I don't see that sort of issue when inspecting the code. 

If setting everything via `SSLParameters` instead of a combination of `SSLParameters` and direct setter calls on the engine will work then it'll be a nice improvement. Our testing should answer that question.

Reply:
`SSLParameters` is basically a POJO for holding properties of the `SSLEngine`. If you look at the `SSLEngine::getSSLParameters` it just constructs a new object with values from the `SSLEngine`. If you look at the `SSLEngine::setSSLParameters` it sets the values back on the `SSLEngine`. The issue in the original codes was the interleaving of values set on `SSLParameters` and `SSLEngine` between the `getSSLParameters` and `setSSLParameters`, effectively wiping out all the direct `SSLEngine` settings.